File: scala/src/io/stpettersen/blackjack/Score.scala
/**
 * Blackjack
 * Copyright 2015 Sam Saint-Pettersen
 * Released under the MIT/X11 License.
 *
 * Scala Swing/AWT implementation.
 */

package io.stpettersen.blackjack
import java.awt.Graphics

/**
 * Score implements a score or information box.
 * @param debug Enable debug messages?
 * @param x X position for score box.
 * @param y Y position for score box.
 */
class Score(debug: Boolean, x: Int, y: Int) extends UI {

  private var score: String = ""
  private val posX: Int = x
  private val posY: Int = y
  Debugger.emit(debug, s"Created score counter at $posX,$posY")

  /**
   * Emit a score or similar message.
   * @param message Score or similar message to emit.
   */
  def emit(message: String): Unit = {
    clear
    score = message
  }

  /**
   * Emit the score or other integer value.
   * @param score Score to emit.
   */
  def emit(score: Int): Unit = {
    emit(Integer.toString(score))
  }

  /**
   * Clear the score box.
   */
  override def clear: Unit = {
    score = ""
  }

  /**
   * Draw the score box.
   */
  override def draw(g: Graphics): Unit = {
    super.draw(g)
    g.drawString(score, posX, posY)
  }
}

File: java/io/stpettersen/blackjack/Score.java
/**
 * Blackjack
 * Copyright 2015 Sam Saint-Pettersen
 * Released under the MIT/X11 License.
 * 
 * Java Swing/AWT implementation.
*/

package io.stpettersen.blackjack;
import io.stpettersen.helper.Debugger;
import java.awt.Graphics;

public class Score extends UI
{
    private String score;
    private int posX;
    private int posY;
    
    /**
     * Score implements a score or information box.
     * @param debug Enable debug messages?
     * @param posX X position for score box.
     * @param posY Y position for score box.
    */
    public Score(boolean debug, int posX, int posY)
    {
        this.posX = posX;
        this.posY = posY;
        score = "";
        Debugger.emit(debug, 
        String.format("Created score counter at %d,%d", posX, posY));
    }
    
    /**
     * Emit a score or similar message.
     * @param message Score or similar message to emit.
    */
    public void emit(String message) 
    {
        clear();
        score = message;
    }
    
    /**
     * Emit a score or other integer value.
     * @param score Score to emit.
    */
    public void emit(int score)
    {
        emit(Integer.toString(score));
    }
    
    /**
     * Clear the score box.
    */
    public void clear()
    {
        score = "";
    }
    
    /**
     * Draw the score box.
     * @param g Graphics object.
    */
    public void draw(Graphics g)
    {
        super.draw(g);
        g.drawString(score, posX, posY);
    }
}
